File: app/src/main/java/com/musicplayer/db/entity/SearchHistory.java
package com.musicplayer.db.entity;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "search_history")
public class SearchHistory {
    @PrimaryKey(autoGenerate = true)
    public int id;
    
    public String keyword;
    public long timestamp;

    public SearchHistory(String keyword, long timestamp) {
        this.keyword = keyword;
        this.timestamp = timestamp;
    }
}

File: app/src/main/java/com/musicplayer/db/entity/SearchHistory.kt
package com.musicplayer.db.entity

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import kotlin.jvm.JvmField

@Entity(
    tableName = "search_history",
    indices = [Index("keyword"), Index("timestamp")]
)
data class SearchHistory(
    @PrimaryKey(autoGenerate = true)
    @JvmField var id: Int = 0,
    @JvmField var keyword: String? = null,
    @JvmField var timestamp: Long = 0
)
